### DISCUSSION TITLE
Select within a create on relational tables?

### DISCUSSION THREAD

**Discussion by nicobrch** [prisma/prisma#14115] [Q&A] [answered]
Hi, I'm trying to select a certain field from another table when creating a new entry on a relational table, but I haven't been able to with prisma 3. I've read [the docs](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations) but the examples haven't helped me, because the IDs are hardcoded.

I have the following model:

```
model badge {
  id            Int             @id @default(autoincrement())
  descripcion   String          @db.VarChar
  image_url     String          @db.VarChar
  torneo        torneo[]
  usuario_badge usuario_badge[]
}

model usuario_badge {
  id         Int     @id @default(autoincrement())
  user_id    Int
  badge_id   Int
  awarded_at String  @db.VarChar
  badge      badge   @relation(fields: [badge_id], references: [id], onDelete: Cascade, onUpdate: NoAction, map: "fk_badge")
  usuario    usuario @relation(fields: [user_id], references: [id], onDelete: Cascade, onUpdate: NoAction, map: "fk_user")
}
```

And I'm trying to replicate this SQL:

```
INSERT INTO usuario_badge (user_id, badge_id, awarded_at) VALUES (1, (SELECT id FROM badge WHERE descripcion = 'value'), '2022-07-02 15:00:00')
```

Basically I'm struggling to do `(SELECT id FROM badge WHERE descripcion = 'value')`, where `value` is a value I know. This is what I've tried so far with no success:

```
await prisma.usuario_badge.create({
                    data: {
                        user_id: data.id,
                        awarded_at: badge.awarded_at,
                        badge_id: {
                            badge: {
                                where: { badge.descripcion = description}
                                select: {
                                    id: true
                                }
                            }
                        }
                    }
                })
```

I'm using prisma 3.15.2 and @prisma/client 4.0.0 on postgres

**Comment by austincrim** [comment, marked as answer]
Hey @nicobrch 👋,

It looks like you are trying to relate an existing `badge` to the new `usuario_badge`, is that right?

If so, take a look at using `connect`:
```ts
await prisma.usuario_badge.create({
  data: {
    user_id: data.id,
    awarded_at: badge.awarded_at,
    badge: {
      connect: { descipcion: description }
    }
  }
})
```

For this query to work, you will also need to add `@unique` to the `badge` model's `descripcion` field.

**Reply by nicobrch** [reply]
Thanks! This is what I needed.
I had to alter the table `badge` to add the unique constraint too, but it worked perfectly.
Tysm!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
